### PR TITLE
Define cases around the service, rhsm_proxy, satellite_multi_org

### PR DESCRIPTION
**Pull request by Junefen**
- [x] test_virtwho_service_after_host_reregister
- [x] test_package_info_with_User_Agent_header
- [x] test_install_uninstall_by_yum: `add steps to check the template.conf`
- [x] test_upgrade_downgrade_by_rpm: `add step to check the virt-who service and configuration after host reboot`
- [x] test_global_options_migration_after_upgrade
- [x] test_rhsm_proxy
- [x] test_rhsm_proxy_in_rhsmconf
- [x] test_rhsm_proxy_in_virtwho_d
- [x] test_non_default_org_with_rhsm_options 
- [x] test_non_default_org_without_rhsm_options

**Test Result**
```
% pytest --disable-warnings -v tests/function/test_service.py -k 'test_virtwho_service_after_host_reregister'
========================================================= test session starts =========================================================
platform darwin -- Python 3.11.2, pytest-7.2.2, pluggy-1.0.0 -- /opt/homebrew/opt/python@3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/yuefliu/workspace/virtwho-test, configfile: pytest.ini
collected 5 items / 4 deselected / 1 selected                                                                                         

tests/function/test_service.py::TestVirtwhoService::test_virtwho_service_after_host_reregister PASSED                           [100%]

======================================= 1 passed, 4 deselected, 2 warnings in 189.22s (0:03:09) =======================================
```

```
 % pytest --disable-warnings -v tests/package/test_info.py -k 'test_package_info_with_User_Agent_header'
========================================================= test session starts =========================================================
platform darwin -- Python 3.11.2, pytest-7.2.2, pluggy-1.0.0 -- /opt/homebrew/opt/python@3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/yuefliu/workspace/virtwho-test, configfile: pytest.ini
collected 6 items / 5 deselected / 1 selected                                                                                         

tests/package/test_info.py::TestVirtwhoPackageInfo::test_package_info_with_User_Agent_header PASSED                             [100%]

================================================== 1 passed, 5 deselected in 44.78s ===================================================

```

```
% pytest --disable-warnings -v tests/package/test_install_uninstall.py -k 'test_install_uninstall_by_yum'
========================================================= test session starts =========================================================
platform darwin -- Python 3.11.2, pytest-7.2.2, pluggy-1.0.0 -- /opt/homebrew/opt/python@3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/yuefliu/workspace/virtwho-test, configfile: pytest.ini
collected 2 items / 1 deselected / 1 selected                                                                                         

tests/package/test_install_uninstall.py::TestInstallUninstall::test_install_uninstall_by_yum PASSED                             [100%]

================================================== 1 passed, 1 deselected in 33.57s ===================================================
```

```
% pytest --disable-warnings -v tests/package/test_upgrade_downgrade.py -k 'test_upgrade_downgrade_by_rpm'
========================================================= test session starts =========================================================
platform darwin -- Python 3.11.2, pytest-7.2.2, pluggy-1.0.0 -- /opt/homebrew/opt/python@3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/yuefliu/workspace/virtwho-test, configfile: pytest.ini
collected 3 items / 2 deselected / 1 selected                                                                                         

tests/package/test_upgrade_downgrade.py::TestUpgradeDowngrade::test_upgrade_downgrade_by_rpm PASSED                             [100%]

======================================= 1 passed, 2 deselected, 2 warnings in 114.79s (0:01:54) =======================================
```
```
% pytest --disable-warnings -v tests/package/test_upgrade_downgrade.py -k 'test_global_options_migration_after_upgrade'
========================================================= test session starts =========================================================
platform darwin -- Python 3.11.2, pytest-7.2.2, pluggy-1.0.0 -- /opt/homebrew/opt/python@3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/yuefliu/workspace/virtwho-test, configfile: pytest.ini
collected 3 items / 2 deselected / 1 selected                                                                                         

tests/package/test_upgrade_downgrade.py::TestUpgradeDowngrade::test_global_options_migration_after_upgrade PASSED               [100%]

============================================= 1 passed, 2 deselected, 1 warning in 59.84s =========================================
```

```
% pytest --disable-warnings -v tests/subscription/test_rhsm_option.py -k 'test_rhsm_proxy'           
========================================================= test session starts =========================================================
platform darwin -- Python 3.11.2, pytest-7.2.2, pluggy-1.0.0 -- /opt/homebrew/opt/python@3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/yuefliu/workspace/virtwho-test, configfile: pytest.ini
collected 12 items / 9 deselected / 3 selected                                                                                        

tests/subscription/test_rhsm_option.py::TestSubscriptionPositive::test_rhsm_proxy PASSED                                        [ 33%]
tests/subscription/test_rhsm_option.py::TestSubscriptionNegative::test_rhsm_proxy_in_rhsmconf PASSED                            [ 66%]
tests/subscription/test_rhsm_option.py::TestSubscriptionNegative::test_rhsm_proxy_in_virtwho_d PASSED                           [100%]

====================================== 3 passed, 9 deselected, 22 warnings in 973.97s (0:16:13) ====================================
```

```
% pytest --disable-warnings -v tests/subscription/test_satellite.py -k 'test_non_default_org'
========================================================= test session starts =========================================================
platform darwin -- Python 3.11.2, pytest-7.2.2, pluggy-1.0.0 -- /opt/homebrew/opt/python@3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/yuefliu/workspace/virtwho-test, configfile: pytest.ini
collected 9 items / 7 deselected / 2 selected                                                                                         

tests/subscription/test_satellite.py::TestSatellite::test_non_default_org_with_rhsm_options PASSED                              [ 50%]
tests/subscription/test_satellite.py::TestSatellite::test_non_default_org_without_rhsm_options PASSED                           [100%]

======================================= 2 passed, 7 deselected, 2 warnings in 271.65s (0:04:31) ======================
```

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -4,10 +4,10 @@ markers =
...
 
-    satellite: cases for satellite
-    rhsm: cases for candlepin
+    notSatellite: cases not for satellite
+    notRHSM: cases not for candlepin
 
-    rhel8: cases only for rhel 8
-    rhel9: cases only for rhel 9
+    notRHEL8: cases not for rhel8
+    notRHEL9: cases not for rhel9
```
Comment (hkx303) [inline pytest.ini:12]: All cases need to update this...

File: tests/function/test_service.py [hunk 2]
```diff
@@ -91,3 +95,70 @@ def test_force_reload(self, virtwho):
         _, _ = virtwho.operate_service(action='force-reload')
```
Comment (hkx303) [inline tests/function/test_service.py:95]: actually, we can just use 
`virtwho.operate_service(action='force-reload')` here

Comment (hkx303) [reply tests/function/test_service.py:95]: but it is fine we can improve this in the future

File: pytest.ini (hunk 1 above)
Comment (Junefen) [reply pytest.ini:12]: yes, and I think we can maintain the marks in cases at last in another single pr.

**Review by hkx303 — APPROVED** [review]

**Pull request merged**